### PR TITLE
add a dart2js bot

### PR DESCRIPTION
**Pull request by devoncarew**
- add another bot that compiles the app and runs our tests using dart2js
- fix https://github.com/flutter/devtools/issues/198
- moved some tasks into a new, 4th bot, for maybe a minute faster cycle times; fix https://github.com/flutter/devtools/issues/168 for now

This did not work for me locally - I want to see what the results look like on the bots.

### CODE REVIEW

**Comment by devoncarew** [comment]
In a good news bad news situation, the new bot passed, but I think that means it was not running the app compiled via dart2js.

**Comment by devoncarew** [comment]
I split the bots out into a 'main' bot, and one each to run our tests compiled w/ dart2js and ddc, and a separate bot to run the tests that require a flutter sdk to be installed.

**Comment by devoncarew** [comment]
I'm happy w/ the refactor now; blocked on https://github.com/flutter/devtools/issues/193.

**Comment by devoncarew** [comment]
Disable the dart2js bot; we can re-enable once the issue w/ the release version of the app is resolved.

File: .travis.yml [hunk 1]
```diff
@@ -13,7 +13,9 @@ addons:
 
 env:
   - BOT=main
-  - BOT=flutter
+  - BOT=test_ddc
+#  - BOT=test_dart2js
```
Comment (jacob314) [inline .travis.yml]: nit: a todo on why this is commented out?

**Pull request merged**